### PR TITLE
Social channel preview

### PR DESCRIPTION
**Pull request by ambaldwin**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: examples/website-demos/social-messaging/package.json [hunk 1]
```diff
@@ -13,6 +13,8 @@
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
     "react-scripts": "4.0.3",
+    "stream-chat": "^3.11.0",
+    "stream-chat-react": "^6.4.1",
```
Comment (DanC5) [inline examples/website-demos/social-messaging/package.json:17]: for now yarn link react, react-dom, and stream-chat-react

File: examples/website-demos/social-messaging/src/App.tsx [hunk 2]
```diff
@@ -1,17 +1,106 @@
...
+import { SocialMessageInput } from './components/MessageInput/SocialMessageInput';
+
+const apiKey = process.env.REACT_APP_STREAM_KEY;
+const user = process.env.REACT_APP_USER_ID;
+const userToken = process.env.REACT_APP_USER_TOKEN;
+
+const userToConnect: { id: string; name?: string; image?: string } = {
+  id: user!,
+  name: user!,
+  image: 'https://ca.slack-edge.com/T02RM6X6B-U01J8HMLA4F-d7bab110afb4-512',
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/App.tsx:26]: save this as an env so it'll work when we build that one

File: examples/website-demos/social-messaging/src/App.tsx [hunk 3]
```diff
@@ -1,17 +1,106 @@
...
-     <ChannelContainer />
+    <div className='chat-container'>
+      <Chat client={chatClient}>
+        <div className={`channel-list-container ${isSideDrawerOpen ? 'sideDrawerOpen' : ''}`}>
+          <SocialChannelListHeader  { ...{ isNewChat, isSideDrawerOpen, setNewChat, setSideDrawerOpen}} />
+          <ChannelList
+            filters={filters}
+            List={(props) => (
+              <SocialChannelList {...props} />
+            )}
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/App.tsx:90]: can shorthand this one

File: examples/website-demos/social-messaging/src/App.tsx [hunk 4]
```diff
@@ -1,17 +1,106 @@
...
+export type MessageType = {};
+export type ReactionType = {};
+export type UserType = { image?: string };
 
 function App() {
+  const [chatClient, setChatClient] = useState<StreamChat | null>(null);
+  const [isListChats, setListChats] = useState(false);
+  const [isListMentions, setListMentions] = useState(false);
+  const [isSideDrawerOpen, setSideDrawerOpen] = useState(false);
+  const [isNewChat, setNewChat] = useState(false);
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/App.tsx:52]: might be easier to put all these state hooks in a context so you don't have to worry about all the props passing

File: examples/website-demos/social-messaging/src/App.tsx [hunk 5]
```diff
@@ -1,17 +1,106 @@
...
+
+const userToConnect: { id: string; name?: string; image?: string } = {
+  id: user!,
+  name: user!,
+  image: 'https://ca.slack-edge.com/T02RM6X6B-U01J8HMLA4F-d7bab110afb4-512',
+};
+
+const filters = { type: 'messaging', members: { $in: [user!] } };
+
+const options = { state: true, watch: true, presence: true, limit: 8 };
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/App.tsx:31]: some of these default to true, not sure which ones tbh, so would be nice if you could find out then we wouldn't need to include

File: examples/website-demos/social-messaging/src/assets/ArrowLeft.tsx [hunk 6]
```diff
@@ -1,5 +1,24 @@
-export const ArrowLeft: React.FC = () => (
-    <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
-<path d="M16 6.99999V8.99999H3.98955L9.49477 14.5052L8.08057 15.9194L0.161133 7.99999L8.08057 0.0805664L9.49477 1.49478L3.98955 6.99999H16Z" fill="black"/>
-</svg>
+type Props = {
+    isNewChat: boolean;
+    setNewChat: React.Dispatch<React.SetStateAction<boolean>>;
+};
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/assets/ArrowLeft.tsx:4]: for all the icons, would prob be easier just to pull these from an AppContext or something like that, you can see how we did that in the event app

File: examples/website-demos/social-messaging/src/components/ChannelContainer/ChannelContainer.tsx [hunk 7]
```diff
@@ -1,18 +1,36 @@
...
-type Props = {};
+type Props = {
+  isNewChat: boolean;
+  setNewChat: React.Dispatch<React.SetStateAction<boolean>>,
+};
 
 export const ChannelContainer: React.FC<Props> = (props) => {
+  const { isNewChat, setNewChat } = props;
+  
+  const ChannelContents = () => {
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/components/ChannelContainer/ChannelContainer.tsx:19]: does this need to be broken out into it's own component?

File: examples/website-demos/social-messaging/src/components/ChannelPreview/utils.tsx [hunk 8]
```diff
@@ -0,0 +1,77 @@
+import type { Channel, ChannelMemberResponse } from 'stream-chat';
+
+import { Avatar } from 'stream-chat-react';
+
+export const AvatarGroup = ({ members }: { members: ChannelMemberResponse[] }) => {
+  if (members.length === 1) {
+    return <Avatar image={members[0].user?.image as string || ''} size={56} />;
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/components/ChannelPreview/utils.tsx:7]: i recently went back and added to the old social app, but you should put a name field on here so if there's no image we at least display the initial on the blue background

File: examples/website-demos/social-messaging/src/components/MessageList/SocialMessageList.tsx [hunk 9]
```diff
@@ -1,13 +1,15 @@
 import React from 'react';
 
+import { MessageList } from 'stream-chat-react';
+
 import './SocialMessageList.scss';
 
 type Props = {};
 
 export const SocialMessageList: React.FC<Props> = (props) => {
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/components/MessageList/SocialMessageList.tsx:9]: i see this just a placeholder but we shouldn't need a message list wrapper component, unless there's an upcoming feature i'm missing

File: examples/website-demos/social-messaging/src/App.tsx (hunk 5 above)
Comment (ambaldwin) [reply examples/website-demos/social-messaging/src/App.tsx:31]: https://getstream.io/chat/docs/javascript/query_channels/
Looks like true is the default for state and watch, presence doesn't appear to be a thing, so I removed.

File: examples/website-demos/social-messaging/src/components/ChannelContainer/ChannelContainer.tsx (hunk 7 above)
Comment (ambaldwin) [reply examples/website-demos/social-messaging/src/components/ChannelContainer/ChannelContainer.tsx:19]: Nope, I can change this. I originally was going to have multiple headers here and was trying to keep it clean, but instead I'm dynamically changing the title of the one header.

File: examples/website-demos/social-messaging/src/components/MessageList/SocialMessageList.tsx (hunk 9 above)
Comment (ambaldwin) [reply examples/website-demos/social-messaging/src/components/MessageList/SocialMessageList.tsx:9]: You're not missing anything, that was just a placeholder initially. Will remove!

File: examples/website-demos/social-messaging/src/assets/ArrowLeft.tsx (hunk 6 above)
Comment (ambaldwin) [reply examples/website-demos/social-messaging/src/assets/ArrowLeft.tsx:4]: Is this implementation in the event app pushed up to the website-demos branch?

File: examples/website-demos/social-messaging/package.json (hunk 1 above)
Comment (DanC5) [reply examples/website-demos/social-messaging/package.json]: i see you resolved this but looks like they still not linked?

Comment (DanC5) [reply examples/website-demos/social-messaging/package.json]: also the TS version is old so can bump to latest and remove the carets on all em

File: examples/website-demos/social-messaging/src/assets/HamburgerIcon.tsx [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+            }}
+            style={{ cursor: 'pointer', marginLeft: '16px' }}
+            xmlns="http://www.w3.org/2000/svg">
+                <path
+                    d="M0 1C0 0.447715 0.447715 0 1 0H17C17.5523 0 18 0.447715 18 1C18 1.55228 17.5523 2 17 2H1C0.447716 2 0 1.55228 0 1ZM0 8C0 7.44772 0.447715 7 1 7H17C17.5523 7 18 7.44772 18 8C18 8.55229 17.5523 9 17 9H1C0.447716 9 0 8.55229 0 8ZM0 15C0 14.4477 0.447715 14 1 14H17C17.5523 14 18 14.4477 18 15C18 15.5523 17.5523 16 17 16H1C0.447716 16 0 15.5523 0 15Z"
+                    fill="black"
+                />
+        </svg>
+    )
+};
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/assets/HamburgerIcon.tsx:25]: looks like you need to adjust your prettier settings to add a newline at end of file like our other repos

File: examples/website-demos/social-messaging/src/assets/NewChat.tsx [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+            onClick={() => {
+                if (!isNewChat) {
+                    setNewChat(true);
+                }
+            }}
+            style={{ borderRadius: '20px', cursor: 'pointer', marginRight: '8px' }}
+            xmlns="http://www.w3.org/2000/svg">
+            <path
+                d="M0 20C0 8.95431 8.95431 0 20 0C31.0457 0 40 8.95431 40 20C40 31.0457 31.0457 40 20 40C8.95431 40 0 31.0457 0 20Z"
+                fill="#005FFF"
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/assets/NewChat.tsx:21]: confirming none of these fills need to be variables?

File: examples/website-demos/social-messaging/src/components/ChatContainer/ChatContainer.tsx [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+
+import { useViewContext } from '../../contexts/ViewContext';
+
+import './ChatContainer.scss';
+
+const apiKey = process.env.REACT_APP_STREAM_KEY;
+const user = process.env.REACT_APP_USER_ID;
+const userImage = process.env.USER_IMAGE;
+const userToken = process.env.REACT_APP_USER_TOKEN;
+
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/components/ChatContainer/ChatContainer.tsx:21]: gonna need all the url params stuff from the current social app, including the weird shit i had to add for the android/ios teams

File: examples/website-demos/social-messaging/src/components/ChatContainer/ChatContainer.tsx [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+const userToConnect: { id: string; name?: string; image?: string } = {
+  id: user!,
+  name: user!,
+  image: userImage,
+};
+
+const filters = { type: 'messaging', members: { $in: [user!] } };
+const options = { message_limit: 20 };
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/components/ChatContainer/ChatContainer.tsx:29]: let's make limit larger than default, not smaller, to reduce load more renders, maybe call it 30

File: examples/website-demos/social-messaging/src/components/SocialChannelList/SocialChannelList.tsx [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+import { ChannelListMessengerProps } from 'stream-chat-react';
+
+import { SocialChannelListFooter } from '../SocialChannelList/SocialChannelListFooter';
+
+import './SocialChannelList.scss';
+
+type Props = ChannelListMessengerProps;
+
+export const SocialChannelList: React.FC<Props> = (props) => {
+  const { children } = props;
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/components/SocialChannelList/SocialChannelList.tsx:12]: will need loading (skeleton like current) and error states eventually, but all good for now

**Review by DanC5 — APPROVED** [review]

File: examples/website-demos/social-messaging/src/assets/NewChat.tsx (hunk 11 above)
Comment (ambaldwin) [reply examples/website-demos/social-messaging/src/assets/NewChat.tsx:21]: updated!

File: examples/website-demos/social-messaging/package.json (hunk 1 above)
Comment (ambaldwin) [reply examples/website-demos/social-messaging/package.json]: I'm tabling this for now. It's not compiling

**Pull request merged**